src/Utilities.h

```diff
@@ -28,6 +38,40 @@ double sse_calc(const std::vector<T>& simval, const std::vector<T>& obsval){
 	
 }
 
+/**
+ * @brief Create new vector summating every Nth element
+ * 
+ * Summate values at every Nth index, e.g. n = 7 would summate
+ * daily data to be weekly data
+ * 
+ * @param data_vector	Vector of data to be reduced
+ * @param n	Nth value at which summation should occur (i.e. index+1)
+ * 
+ * @return Vector of summation values
+ */
+template<typename T>
+std::vector<T> AccumulateEveryNth(const std::vector<T>& data_vector, const int& n)
+{
+	std::vector<T> _temp = {};
+	T data_val(0);
+
+	for(int i{0}; i < data_vector.size()+1; ++i)
```

Comment:
The `i < data_vector.size() + 1` is very clunky, and it takes several reads of the function before it's obvious how it works. In particular, it is less clear than Thibaud's original code for `weekly`. I'd suggest reverting to something closer to that implementation.